client/src/ts/dd_main.ts

```diff
@@ -0,0 +1,120 @@
+const xTimeBuffer = 1000;
+var motor_power = new streamGraph.StreamingDataset(
+    motor_data, {"color" : "rgb(88, 86, 214)"});
+motor_power.setFilter(function(data: any[]) {
+  return data.filter(function(datum: any) {
+    return datum.x >= timeDomain.begin() - xTimeBuffer;
+  });
+});
+
+// Plot configuration.
+var streamingPlot = new streamGraph.StreamingPlot(new Plottable.Plots.Area());
+streamingPlot.datasets([ motor_power ]);
+streamingPlot.plot.y(function(d: any) { return d.y }, yScale);
+streamingPlot.x(function(d: any) { return d.x; }, xScale);
+streamingPlot.plot.attr("stroke",
+                        function(d: any, i: any, ds: Plottable.Dataset) {
+                          return ds.metadata().color;
+                        });
+streamingPlot.plot.attr(
+    "fill", function(d: any, i: any, ds: Plottable.Dataset) {
+      return ds.metadata().color.replace(")", ", 0.6)").replace("rgb", "rgba");
+    });
+streamingPlot.plot.attr("stroke-width", 3);
+
+// Chart configuration.
+var group = new Plottable.Components.Group([ streamingPlot.plot ]);
+var chart = new Plottable.Components.Table(
+    [ [ yAxis, group ], [ null, xAxis ], [ null, xLabel ] ]);
+chart.renderTo("div#graph");
+
+function UpdatePlot() {
+  timeDomain.domain();
+  streamingPlot.redraw();
+  chart.redraw();
+}
+
+motor_power.dataUpdate = function() { UpdatePlot(); };
+
+function AddData() {
+  const now = Date.now()
+  motor_power.addData({x : now, y : Math.cos(now / 1000)});
+}
+
+window.setInterval(function() { AddData(); }, 100);
+window.addEventListener("resize", function() { chart.redraw(); });
+
+// DIALS
+
+var speedDialOptions = new dial.DialOptions();
+speedDialOptions.rectColor = "#1D1F26";
+speedDialOptions.rectOpacity = 0;
+speedDialOptions.dialColor = "#11b981";
+speedDialOptions.dialOpacity = 1;
+speedDialOptions.font = "Open Sans";
+speedDialOptions.fontColor = "white";
+speedDialOptions.thickness = 30;
+speedDialOptions.angleOffset = 0.5 * Math.PI;
+speedDialOptions.angleArc = 1.5 * Math.PI;
+speedDialOptions.fontSize = 200;
+speedDialOptions.padding = 20;
+
+var socDialOptions = new dial.DialOptions();
+socDialOptions.rectColor = "#1D1F26";
+socDialOptions.rectOpacity = 0;
+socDialOptions.dialColor = "#11b981";
+socDialOptions.dialOpacity = 1;
+socDialOptions.thickness = 80;
+socDialOptions.showText = false;
+socDialOptions.angleOffset = 0.5 * Math.PI;
+socDialOptions.angleArc = 1 * Math.PI;
+socDialOptions.rotation = dial.Direction.CounterClockwise;
+var speedDial = new dial.Dial(<HTMLDivElement>document.getElementById("speedometer"),
+                         speedDialOptions);
+var batteryDial =
+    new dial.Dial(<HTMLDivElement>document.getElementById("soc"), socDialOptions);
+
+speedDial.updateValue(100);
+batteryDial.updateValue(100);
+
+let date = new Date();
+document.getElementById("status").innerHTML = date.toLocaleTimeString();
+
+//window.setInterval(function() {
```

Comment:
This is disabled right now but it is a good demo if you want to see the dial move so I kept it in. Also is a good example of what to do when we connect it to the websocket.